invoices/invoiceregistry.go

```diff
@@ -449,8 +461,17 @@ func (i *InvoiceRegistry) LookupInvoice(rHash lntypes.Hash) (channeldb.Invoice,
 // NotifyExitHopHtlc attempts to mark an invoice as settled. If the invoice is a
 // debug invoice, then this method is a noop as debug invoices are never fully
 // settled. The return value describes how the htlc should be resolved.
+//
+// When the preimage of the invoice is not yet known (hodl invoice), this
+// function moves the invoice to the accepted state. When SettleHoldInvoice is
+// called later, a resolution message will be send back to the caller via the
+// provided hodlChan. Invoice registry sends on this channel what action needs
+// to be taken on the htlc (settle or cancel). The caller needs to ensure that
+// the channel is either buffered or received on from another goroutine to
+// prevent deadlock.
 func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
-	amtPaid lnwire.MilliSatoshi) (*HodlEvent, error) {
+	amtPaid lnwire.MilliSatoshi, hodlChan chan<- interface{}) (
+	*HodlEvent, error) {
```

Comment:
The notification channel can be returned _within_ the `HodlEvent` response. The registry can handle buffering it properly, and the client then checks directly or indirectly (via the method) if there's a delayed response/future.

Reply:
The `hodlChan` argument is there, because I want all HodlEvents to be sent on the same channel. So that the link only needs to listen on that single channel in its main event loop. If every call to `NotifyExitHopHtlc` would yield a new channel, how to listen to them all in a clean way?

Reply:
Ahh I see what you're getting at here: a single channel with multiple events sent over it, rather than needing to have a new channel for even hodl event.

Reply:
Yes that indeed. Otherwise we'd get that reflection-based select over a list.